template/template.go

```diff
@@ -133,6 +133,24 @@ func (p *PostProcessor) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
+// If Post Processor Is not in this list it means that it's valid for all builders
+var validBuildersPerPostProcessor = map[string][]string{
+	"vsphere":          {"vmware-iso", "vmware-vmx"},
+	"vsphere-template": {"vmware-iso", "vmware-vmx"},
+}
```

Comment:
Super nice !

Reply:
This is much more complex than I thought! I can understand just by reading what you say and it makes sense to me.  
About this one I've implemented, It wasn't clear to me if this PR is supposed to be merged or not since it won't solve the bigger picture of the issue I was trying to solve.